lib/oscap_tests.pm

```diff
@@ -41,53 +59,367 @@ our $f_vlevel = 'ERROR';
+# If set to 1 - tests will use files from scap-security-guide package
+# If set to 2 - tests will use files from compliance-as-code-compiled repository - https://gitlab.suse.de/seccert-public/compliance-as-code-compiled
+# If set to 3 - tests will use files cloned and built from ComplianceAsCode repository master branch - https://github.com/ComplianceAsCode/content.git
+our $use_content_type = 1;
+
+# Option configures to use or not functionality to remove from DS and ansible file rules for which do not have remediations
+# If set to 1 - rules for which do not have remediations will be removed from DS and ansible file rules for which do not have remediations
+# If set to 0 - no changes done.
+our $remove_rules_missing_fixes = 1;
+
+# Option configures to use or not functionality to exclude rules for profiles defined in file:
+# https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/blob/main/content/openqa_tests_exclusions.yaml
+# If set to 1 - rules defined in exclusions files are excluded for remediation and evaluation
+# If set to 0 - rules defined in exclusions files are not used
+our $use_exclusions = 1;
+
+# Keeps count of reboots to control it
+our $reboot_count = 0;
+
+# evaluate execution count done by test execution. Set in oscap_security_guide_setup.pm
+# and "security/oscap_stig/oscap_xccdf_eval" need to be set in the schedule yaml file accordingly
+our $evaluate_count = 2;
+
+# Stores CCE IDs of failed ansible remediation tasks.
+# Used in second ansible remediation as exclusions.
+our $failed_cce_ids_ref;
+
+# List to collect needed run results
+our @test_run_report = ();
+our $test_run_report_name = "test_run_report.txt";
+
+# Get sle version "sle12" or "sle15"
+our $sle_version = '';
 
 # Upload HTML report by default
 set_var('UPLOAD_REPORT_HTML', 1);
 
 # Set value for 'scap-security-guide' ds file
 sub set_ds_file {
-
     # Set the ds file for separate product, e.g.,
     # for SLE15 the ds file is "ssg-sle15-ds.xml";
     # for SLE12 the ds file is "ssg-sle12-ds.xml";
     # for Tumbleweed the ds file is "ssg-opensuse-ds.xml"
     my $version = get_required_var('VERSION') =~ s/([0-9]+).*/$1/r;
     $f_ssg_sle_ds =
       '/usr/share/xml/scap/ssg/content/ssg-sle' . "$version" . '-ds.xml';
+    $f_ssg_sle_xccdf =
+      '/usr/share/xml/scap/ssg/content/ssg-sle' . "$version" . '-xccdf.xml';
+}
+sub set_ds_file_name {
+    # Set the ds file for separate product, e.g.,
+    # for SLE15 the ds file is "ssg-sle15-ds.xml";
+    # for SLE12 the ds file is "ssg-sle12-ds.xml";
+    # for Tumbleweed the ds file is "ssg-opensuse-ds.xml"
+    my $version = get_required_var('VERSION') =~ s/([0-9]+).*/$1/r;
+    $ssg_sle_ds =
+      'ssg-sle' . "$version" . '-ds.xml';
+    $ssg_sle_xccdf =
+      'ssg-sle' . "$version" . '-xccdf.xml';
 }
 
+sub replace_ds_file {
+    # Replace original ds file whith built or downloaded from repository
+    my $self = $_[0];
```

Comment:
it is more typical to use in Perl ` my ($self) = @_`

Reply:
Fixed